src/cli/tls_server.cpp

```diff
@@ -16,7 +16,14 @@
 #if defined(BOTAN_HAS_TLS) && defined(BOTAN_TARGET_OS_HAS_FILESYSTEM) && \
    defined(BOTAN_TARGET_OS_HAS_SOCKETS)
 
-#if defined(SO_USER_COOKIE) || defined(SO_RTABLE)
+#if defined(SO_MARK) || defined(SO_USER_COOKIE) || defined(SO_RTABLE)
+#if defined(SO_MARK)
+#define BOTAN_SO_SOCKETID SO_MARK
+#elif defined(SO_USER_COOKIE)
+#define BOTAN_SO_SOCKETID SO_USER_COOKIE
+#else
+#define BOTAN_SO_SOCKETID SO_RTABLE
+#endif
 #define SOCKET_ID 1
```

Comment:
Can we remove the definition of `SOCKET_ID` and just use the existence of `BOTAN_SO_SOCKETID` to decided if we should call `setsockopt` or not?

Reply:
valid point

Reply:
Looks good, thanks!